src/Controls/src/Core/DateTimeTypeConverter.cs

```diff
@@ -0,0 +1,104 @@
+#if NET6_0_OR_GREATER
+using System;
+using System.ComponentModel;
+using System.Globalization;
+using Microsoft.Maui.Controls.Xaml;
+
+namespace Microsoft.Maui.Controls;
+
+internal class DateTimeTypeConverter : TypeConverter
+{
+    public override bool CanConvertFrom(ITypeDescriptorContext? context, Type? sourceType)
+        => sourceType == typeof(DateTime) || sourceType == typeof(DateOnly);
```

Comment:
DateTime should only convert To DateTime, From DateOnly

Reply:
I've updated the changes to support conversion only from DateOnly to DateTime.